gpflow/experimental/check_shapes/config.py

```diff
@@ -58,10 +96,18 @@ def set_enable_check_shapes(enabled: bool) -> None:
     See also :func:`disable_check_shapes`.
     """
     global _enabled
-    _enabled = enabled
+    if enabled == True:  # pylint: disable=singleton-comparison
```

Comment:
> Any reason to use `==` rather than `is`? If it's to handle numpy booleans then wouldn't that not typecheck anyway, since `numpy.bool_` is not a subtype of `bool`?

I don't think one should be using `is` with `bool`s. I cannot find anywhere that guarantees that it works? AFAICT it mostly works because CPython usually cashes the `True` and `False` instances?